File: week2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    // int sentences = count_sentences(text);

    // Compute the Coleman-Liau index

    printf("%d lettes count\n", letters);

    // Print the grade level
}

int count_letters(string text)
{
    int letters_count = 0;
    // Return the number of letters in text
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // printf("%c is ",text[i]);
        if (isalpha(text[i]) == 0)
        {
            // printf("not");
        }
        else{
            letters_count ++;
            // printf("a");
        }
        // printf(" letter\n");
    }

    return letters_count;
}

int count_words(string text)
{
    int words_count = 0;
    // Return the number of words in text
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // letter found
        if (isalpha(text[i]) != 0)
        {

        }
        // space found
        else if (isspace(text[i]) !=0)
        {

        }
    }
    return words_count;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    return 1;
}

File: week6/sentimental-readability/readability.py
import math

def count_letters(text):
    letter_count = 0
    for char in text:
        if char.isalpha():
            letter_count += 1
    return letter_count

def count_words(text):
    words_count = text.count(' ') + 1
    return words_count

def count_sentences(text):
    punctuation_count = 0
    for char in text:
        if char in ['.', '!', '?']:
            punctuation_count += 1
    return punctuation_count

def coleman_liau_index(L, S):
    return 0.0588 * L - 0.296 * S - 15.8

def rounding_up(x):
    fractional_part = x - math.floor(x)
    if fractional_part >= 0.5:
        return math.ceil(x)
    else:
        return math.floor(x)

text = input("Text: ")

# Count the number of letters, words, and sentences in the text
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

# print(f"letters: {letters} words: {words} sentences: {sentences}")

L = (letters / words) * 100
S = (sentences / words) * 100

# Compute the Coleman-Liau index
index = coleman_liau_index(L, S)
# print(f"L: {L} S: {S} Index: {index}")
# Rounding
grade = rounding_up(index)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade", int(grade))
